module6/lessons/Week4/StacksAndQueues.md

```diff
@@ -0,0 +1,57 @@
+---
+layout: page
+title: Stacks and Queues - CS Topic - Data Structure
+---
+
+## Introduction
+
+Today's lesson on Stacks and Queues will be a mixture of independent research, class walkthrough and discussion, and diving into code implementations.
+
+### Independent Learning
+Start by taking 30 minutes to independently research the Stack and Queue Data Structures.
+
+I recommend that you start by reading these two articles:
+* [Stacks and Overflows](https://medium.com/basecs/stacks-and-overflows-dbcf7854dc67#.3l76d12dq)
+* [Difference between Stacks and Queues](https://www.geeksforgeeks.org/difference-between-stack-and-queue-data-structures/)
+
+As you're researching, look for the answers to the following questions.
+
+1. What are the key operations that can be done on a Stack, what about for a Queue?
+1. What is similar between Stacks and Queues?
+1. What is different between Stacks and Queues?
+1. What are examples of Stacks and Queues "in the wild"?
+
+Also, start thinking about how you would use a LinkedList or an Array to build a Stack or a Queue. We will dig more into implementation in the next section!
+
+## Walkthroughs and Code
+### Stacks
+
+The Stack Data Structure can be implemented using either an Array or a Linked List. For today, we are going to implement a Stack using an Array and it will overflow if the Stack is larger than 10.
```

Comment:
I wonder if this is a good time to revisit the fundamentals and talk about _why_ it could be an array or linked list and the importance of order/indexing

Reply:
Ooo good call. I think I'll add a quick review of arrays and linked lists and why these tools are helpful here.